### ISSUE TITLE
Preview of v3

### ISSUE THREAD

**Issue by fancyapps** [fancyapps/fancyBox#1169]
Hi,

You can take a look at upcoming fancyBox3 here - [http://fancyapps.com/fancybox/3/
](http://fancyapps.com/fancybox/3/
)

Putting the finishing touches right now and later will start working on homepage and documentation. There will be a lot to cover as the script has changed completely (for example, it is possible to show more than one modal at a time).

Thanks!

**Comment by mjcampagna** [comment]
Looking very promising. I'll be eager to get my hands on the final version and to put it through its paces.

**Comment by TJung1968** [comment]
Looks great, and works great on touch devices! Can't wait to see the final version! :-)

**Comment by overtune** [comment]
Looks really great! Nice work!

Is there any plans of sometime remove the jQuery dependency?
As the web has evolved today I see less and less need for jQuery.
That would also make it much easier to implement/use Fancybox with other frameworks such as React/Angular.

But again, really nice work!

**Comment by fancyapps** [comment]
Thanks. 

v3 will still be using jQuery. I was just thinking about that lately and decided that v4 will be standalone.

**Comment by acidcrash2005** [comment]
It is awesome! Very cool features! I want to have a try on my projects!

**Comment by carlos3duardo** [comment]
Great! I can hardly wait to use this.
It's so much better and beauty than before. Congrats!

**Comment by evgene** [comment]
can u put button print in right top button panel?

**Comment by fancyapps** [comment]
Even if there will not be print button by default, you will be able to easily create your own.

**Comment by evgene** [comment]
I newbie, perfect if you write example in new site, how do print button. Thank you.

**Comment by mjcampagna** [comment]
The site is looking good, but the download button isn't hooked up to the downloadable file. An oversight, or have you not launched yet?

**Comment by redpik** [comment]
great news @fancyapps ! looks great.
Can you create a git branch for v3?

**Comment by fancyapps** [comment]
Thanks! @mjcampagna @redpik I`m right now testing and cleaning up the code and later will push to GitHub. Then download button will start working :)
@evgene yeah, I`ll make a tutorial

**Comment by robfaas** [comment]
I use Fancybox on photography sites. It would be great if it is possible to copy-protect the images. I know it always will be possible to copy by printscreen, but still it will prevent a lot of copying.

**Comment by fancyapps** [comment]
@robfaas You know what, I actually implemented a simple right-click protection. It is disabled by default, and can be enabled like this:

```
$(".fancybox").fancybox({
	image : {
		protect : true
	}
});
```

**Comment by mromid** [comment]
![image](https://cloud.githubusercontent.com/assets/22810402/21576659/64514e2e-cf4d-11e6-8302-32bab3317e54.png)

Thank you for this plugin ,
first , look at the my image , i think problem is z-index
secend after zoom + , i cant see zoom - button


end sorry for my bad english

**Comment by fancyapps** [comment]
1) I increased z-index values
2) So, you really think there should be a "zoom-out" button? Single click (or double tap on touch devices) would zoom image out.

Thanks.

**Comment by Reda-Ahmad** [comment]
I want to say fancybox2 was awesome and cant wait to work with v3, any expected release date plz !

**Comment by mijatovic85** [comment]
Looks great, is it safe to use the beta version at this point? I want this swipe option on mobile devices and to disable arrow also on mobile devices, because unfortunately, this option is not supported in version 2?

**Comment by fancyapps** [comment]
Yes, it is ready, working on documentation right now.

**Comment by nikas64527** [comment]
Hey, it's awesome! Thanks, it looks so great!
But docs page returns 404 error, cant use it :(

P.S.
Sorry for my English (not native).

**Comment by enenemultimedia** [comment]
I haven't seen anything like this in a long time. 
Can't wait to get my hands dirty with this.

Cheers!

**Comment by mjcampagna** [comment]
Hi Janis,

I've pulled the Fancybox 3 files from your demo site so that I could start messing around with them in my current project, being locally developed, and I'm having a small issue at mobile display sizes. I though you should be aware of it, in case it's something you can fix before full release. I've got this CSS in my basic site styling, standard rules for responsive image handling:

`img {
  height: auto;
  max-width: 100%;
}`

So when I load an image in Fancybox at mobile display size, the image is smaller than it should be, and appears in the top-left corner of the display. If I disable these rules, then Fancybox seems to display the image as it should.

As a safety precaution, Fancybox 3's CSS should probably do a reset in its own stylesheet. This seems to do the trick, though maybe you'll want to implement it in a different way:

`.fancybox-slide img {
  max-width: none;
}`

Cheers,
Matt

**Comment by fancyapps** [comment]
@mjcampagna Thank you, I really appreciate your help. I did include a fix in the stylesheet.

**Comment by mjcampagna** [comment]
My pleasure to help. I've been using Photoswipe for galleries and Magnific Popup for other modal content. Honestly, was never keen on Fancybox 2 -- before the mobile/touch era, Highslide was my jam -- but this latest version is fantastic, and I'm keen to replace the Photoswipe/Magnific combo with a single, more capable library. Fancybox 3 has the potential to improve my application greatly, so anything I can do to help you make it better, I'm in. =D

**Comment by phifa** [comment]
This looks amazing. Can we use this version or will there be sth else? 
http://fancyapps.com/fancybox/3/fancybox/jquery.fancybox.min.js 

And another question: How does fancybox 3 differ / compare to photoswipeq? maybe @mjcampagna can weigh in on this as well?

**Comment by robfaas** [comment]
On my Windows mobile, when I change the orientation from vertical to horizontal, the overlay is reacting strange; only covering a small part at the top of the screen.

**Comment by mjcampagna** [comment]
@phifa I'll lead off with this line from the Photoswipe documentation:

"PhotoSwipe is not a simple jQuery plugin, at least basic JavaScript knowledge is required to install."

So right off the bat, it's not so easy to start using. I've used and customized Photoswipe extensively, and it's fantastic for what it does; but what it does is limited. It's purpose is in the name. It only supports images, and is not a general lightbox. So no inline content, iframe, video, etc.

Using Photoswipe means having to structure your images in a specfic way, or that you have to be well-versed enough with Javascript to write your own init. If you want to customize it, then you always have to edit the script file for the UI. I've done a lot of this, and it's not fun.

I haven't gotten far with Fancybox 3 yet, as I'm still waiting for the final version. But with the tinkering I've done, implementation has been as simple as:

1. including the js
2. including the css
3. adding "data-fancybox" to my elements

It looks like it can be further customized by writing your own JS, but if you want to run it vanilla, you don't even have to initialize the script. It just works, without writing a single line of JS. Frankly, I am amazed at just how simple it is.

Also, Fancybox 3 supports content of all kinds, and you can load content on top of content. So, for example, I can load an image and then call a second popup on top of the image containing its EXIF data. I'm looking forward to playing with such possibilities.

Using Photoswipe, I've been using Magnific Popup (from the same author) for non-gallery modal content. To use the same example, I'm looking at an image, and then click a custom button to view the EXIF data; that button closes Photoswipe, then loads a separate popup of inline content to show the EXIF data using Magnific Popup. It's roundabout, and not great having to move between modals and scripts like this.

**Comment by fancyapps** [comment]
Nice summary, @mjcampagna, thanks!
Of course, there are other features like automatic content recognition (including YouTube, Google maps and others), "auto-height" for iframes, thumbnails, basic photo copy protection, etc.
It is easy to redesign and create something unique, check out demos at homepage. 
 
I actually have tons of ideas, so, stay tuned!

**Comment by fancyapps** [comment]
@phifa You can start using it, there will be just minor changes.

@robfaas fancyBox3 supports all modern browsers (surprisingly, even IE9), but it has not been tested on Windows Mobile. And it looks like jQuery supports only Android and Safari - https://jquery.com/browser-support/

**Comment by eladnova** [comment]
Hey - looks great.

I've a commercial license. Whats the upgrade path here in terms of cost?

Also, will there be an option to:

   1. Run thumbnails along bottom of screen in a row
   2. Ensure large image doesn't run behind the image title and description
   at the bottom of the window

**Comment by fancyapps** [comment]
@eladnova Thanks!

fancyBox3 is a free upgrade for all license holders. 

About these both options - not yet, but it will be possible in the future. I am thinking about rewriting CSS as SCSS, or maybe creating something like a wizard where you would choose layout, colors, etc.

**Comment by redpik** [comment]
SCSS files, great idea for customizations. Can I help?

**Comment by phifa** [comment]
omg scss would be amazing. yeah, anything we can do? when will it be on bower?

**Comment by phifa** [comment]
I installed it and tried it, it kinda works, but I get this weird thing when I click on an image: 
http://recordit.co/pMv2qkSWF3 

The buttons at the top are working great however. 

Windows 10 with Chrome 55 and jquery v2.2.4

**Comment by mjcampagna** [comment]
That's the zooming. Looks to be working as designed. 

Click and drag left or right to mimic swiping (as if on a phone), or use the arrow keys or the navigation buttons in the UI to advance. 

What type of response were you expecting?

**Comment by phifa** [comment]
Ah got it. It is because of jQuery  2.2.4. When I use 3.1 it works as expected. Too bad my projects still use jquery 2

**Comment by AdamST94** [comment]
I'm really looking forward to this. I can't wait to implement Fancybox 3 into my website. I really like the new image gallery and the ability to play youtube videos! Hope the set-up is straight forward like fancybox 2! Keep up the amazing work.

**Comment by mjcampagna** [comment]
@fancyapps, @phifa Regarding jQuery, I've only used Fancybox 3 with jQuery 3 so far, as it's what I use for most of my projects. The notable exception here is Wordpress, which currently utilizes jQuery 1.12.4. And as I've learned the hard way, it's usually not a good idea to replace it with a newer version.

So, does Fancybox 3 require jQuery 3? Will there be any way of having it work with older jQuery versions, such as when included in Wordpress themes?

**Comment by redpik** [comment]
@mjcampagna 
"it's usually not a good idea to replace it with a newer version"
nope, you can "deregister" jquery 1.x and load another jquery.
example here https://github.com/redpik/Twoobl/blob/master/lib/scripts.php#L17

**Comment by mjcampagna** [comment]
@redpik
Yes, I'm aware of such techniques and have used them in the past. The result, though, was customers attempting to use third-party plugins which broke because of the jQuery version, and so it created plugin compatibility and support issues. It's for reasons such as this that, in Wordpress circles, it's bad practice to deregister Wordpress' packed-in jQuery.

**Comment by fengwei77** [comment]
look like great!
when will release first version!!
want taste it!

**Comment by danielnr87** [comment]
Looking forward to this - much improved and modernised. FancyBox is still my preferred lightbox. Lets hope it arrives very soon! (i've got projects I wanna use it in :))

**Comment by mijatovic85** [comment]
I have one issue, can you help me how to fix when I try to scroll on mobile device, image is full width of the screen, just little edges left and right [http://prntscr.com/dtgr14](url) 

Please take a look on screenshot. When I try to scroll down in this situation, image opens, instead of scrolling content down. How to fix this issue?

**Comment by fancyapps** [comment]
Hmm, does your page has something like - `<meta name="viewport" content="user-scalable=no, width=device-width, initial-scale=1.0, maximum-scale=1.0">` ?

Anyway, I replaced "touchstart" to "touchend" event that triggers opening, maybe that would help, too.

Well, looks like tomorrow will be the launch day.

**Comment by mijatovic85** [comment]
Thanks for your reply, and yes, I already have meta name="viewport" in head tag, but it does not change anything, unfortunately. I am not sure about touchend and touchstart, can you be more specific please? I really need your help.

**Comment by adam-shaun** [comment]
Is there anywhere we can download the v3 preview and test it out?

**Comment by fancyapps** [comment]
Well, touchstart was used to remove 300ms tap delay on links triggering fancybox, but I changed to touchend event.

So, grab the latest version from the homepage (js: http://fancyapps.com/fancybox/3/fancybox/jquery.fancybox.min.js and css: http://fancyapps.com/fancybox/3/fancybox/jquery.fancybox.min.css)

**Comment by mijatovic85** [comment]
Thank you for your reply, but unfortunately, the same is happening :( When try to scroll content down by touchswipe image on mobile device, image opens instead of scrolling content down. **I can only scroll by touchswipe on white space arround the images**

I really don't know how to fix this issue. Whether it would solve the problem if I activate tap delay of 300ms, I don't know?

**Comment by fancyapps** [comment]
@mjcampagna could you give a link to your page or recreate this issue on codepen - http://codepen.io/fancyapps/pen/jyEGGG/ ?
btw, do you experience the same issue when viewing http://fancyapps.com/fancybox/3/ ?

**Comment by adam-shaun** [comment]
So all I need to demo this?

jquery.fancybox.min.js 
jquery.fancybox.min.css

Thanks for the help.

**Comment by fancyapps** [comment]
@adam-shaun yes, just two files :)

**Comment by mijatovic85** [comment]
Woow, no I don't have the same experince when viewing http://fancyapps.com/fancybox/3/ on mobile device. I can scroll content by swipedown on image.

**Comment by adam-shaun** [comment]
@fancyapps Thanks!

**Comment by mjcampagna** [comment]
> @mjcampagna could you give a link to your page or recreate this issue on codepen - http://codepen.io/fancyapps/pen/jyEGGG/ ?
btw, do you experience the same issue when viewing http://fancyapps.com/fancybox/3/ ?

@fancyapps, I'm not sure which issue you're referring to. Can you point me to the post to which this is a reply?

**Comment by mijatovic85** [comment]
@fancyapps please tell me what's the difference between image gallery on http://fancyapps.com/fancybox/3/ and js code that you send just before.  Because it works well on mobile and it can be scroll down over the image

**Comment by fancyapps** [comment]
Sorry, @mjcampagna, it was meant for @mijatovic85 !

**Comment by mijatovic85** [comment]
> Woow, no I don't have the same experince when viewing http://fancyapps.com/fancybox/3/ on mobile device. I can scroll content by swipedown on image. What's the difference?

> @fancyapps please tell me what's the difference between image gallery on http://fancyapps.com/fancybox/3/ and js code that you send just before. Because it works well on mobile and it can be scroll down over the image

**Comment by fancyapps** [comment]
There is no difference, it is the same code, therefore I asked for a link to your page to look for a clue, because it works fine for me.

**Comment by mijatovic85** [comment]
[http://mile.x3.rs/apartment/apartment1.html](url)

You will noticed that you can not scroll content down by swipedown over image.

**Comment by fancyapps** [comment]
Your page is not responding for me. btw, what device/browser are you using?

**Comment by jarred-cz** [comment]
@mijatovic85 I can reproduce what you mean. But it has to be something wrong with your code, because original preview on http://fancyapps.com/fancybox/3/ works OK in this case.

**Comment by mijatovic85** [comment]
@fancyapps Google chrome, samsung galaxy s5

**Comment by TJung1968** [comment]
mijatovic85, you don't have the same jquery.fancybox.min.js file that is currently available at  http://fancyapps.com/fancybox/3/  – apparently, some changes were made after you downloaded that file.

**Comment by adam-shaun** [comment]
Hey @mijatovic85 I noticed that when you open a image on your site with the new fancybox that your navigation will jump to the right and then back in place. Is there a fix for this? I have a feeling it might happen when I try and implement fb3 it to my website.

**Comment by mijatovic85** [comment]
@TJung1968 regardless of whether or not, the same happens again. Now I have the same code such as available at http://fancyapps.com/fancybox/3/

@adam-shaun, you will noticed when you open image on my site, vertical scroll bar disappear, then navigation jump little bit to the right, that's ok

**Comment by mijatovic85** [comment]
Guys, I think I'm a bit closer to solution, in my css file I have `body, html {
    height: 100%;
}
`

When I get rid of that line of code , now I can scroll on image too, but after hald second, images over which I swipedown appear.

**Comment by jarred-cz** [comment]
@mijatovic85 its because "touchend" instead of "touchstart" I guess

**Comment by mijatovic85** [comment]
@jarred-cz Thank you for your reply, do you know how to change touchend to touchstart in this situation when we use fancybox?

**Comment by jarred-cz** [comment]
@mijatovic85 its the event which Janis changed few hours ago - its doesnt work with touchstart in your case neither

**Comment by mijatovic85** [comment]
So what do we do then? Could this constitute as a bag or...?

**Comment by adam-shaun** [comment]
@mijatovic85 Is there a fix for the jumping? I expect that I will get the same sort of issue when I add the new Fancybox to my website. 

I had to disable the 'overlay: lock' on Fancybox 2 to stop my navigation from jumping to the right (although this allowed the website behind the image to be scrollable, which caused some issues on iOS and Android).

**Comment by mikhailrepka** [comment]
http://fancyapps.com/fancybox/3/docs/ — not found, download button not working =/ I want to download fancybox3 and test it.

**Comment by mjcampagna** [comment]
@mikhailrepka Fancybox 3 has not been officially released yet. If you want to test it, you can download the files from the preview link's source code. Otherwise, you should wait for the official launch to download the release package.

**Comment by mijatovic85** [comment]
Guys, I want to tell you what it was a mistake, instead of parent div "gallery", I used `a[data-fancybox="gallery"]` 

Wrong `$(document).ready(function() {
  $('a[data-fancybox="gallery"]').fancybox();
  });`

Correct `
$(document).ready(function() {
  $("gallery").fancybox();
  });
`

**Comment by TJung1968** [comment]
On Fancybox 2 (and the old Fancybox 3 beta) I used two things from the "Tips & tricks" section, namely "Use element instead of attribute" as shown in this fiddle: <http://jsfiddle.net/cWNw9/>, and "Add something to the title (add download link)" <http://jsfiddle.net/zAe6Z/>. Both of them don't seem to work anymore in Fancybox 3. Do I have to change something in the code, or are things like this no longer supported at all in Fancybox 3?

**Comment by fancyapps** [comment]
@TJung1968 It is even simpler: http://codepen.io/fancyapps/pen/ygeYGx?editors=1010

**Comment by fancyapps** [comment]
I`m not that good at writing documentation, but I hope this makes sense - http://fancyapps.com/fancybox/3/docs/

@mijatovic85 : try updating to the latest.

@adam-shaun : simply `add compensate-for-scrollbar` to your fixed positioned element (http://fancybox.docs/3.0/#faq)

**Comment by mjcampagna** [comment]
> I'm not that good at writing documentation, but I hope this makes sense...

Looks good go me. I'll be sure to let you know if I have any feedback once I start getting my hands dirty.

**Comment by d3rson** [comment]
Very Nice update from FancyBox2. There is a way to make the Thumbnails be opened by Default ?

**Comment by adam-shaun** [comment]
Hello! I implemented fancybox 3 into my website, however, I am experiencing some issues that I do not know how to fix. 

Firstly, when I click a image or video the navigation and body will jump slightly, even though I have included 'compensate-for-scrollbar' in my navigation class.

Secondly and finally, when clicking an image or video to open fancybox the scroll bar is still visible and you can scroll behind whilst the image / video is up. I have attempted to use the overlay helper and set it to 'lock' but it isn't affecting it. 

I would be really grateful for any help that can be provided.

Example of issues: http://codepen.io/adam94/pen/rjxGxM

I have also set up test pages if you wish to see the problems:
http://adamthomasportfolio.com/a2_test.html
http://adamthomasportfolio.com/d2_test.html
http://adamthomasportfolio.com/p2_test.html

**Comment by mjcampagna** [comment]
@adam-shaun On p2_test.html, my console is showing syntax error on line 222. Looks to me like you need to check your matching pairs of brackets, as I think you've got too many closures.

On d2_tesxt.html, I'm seeing a lot of red in my console. You'll want to resolve these errors.

Do you know how to use the console to troubleshoot errors? Can point you in the right direction if necessary.

**Comment by mjcampagna** [comment]
@fancyapps 

I've now gone through most of the documentation. A few remarks/questions.

1.
You've got a typo here: "Enable questures (tap, zoom, pan and pinch)"; should be "gestures".

2.
The option "infobar" works as described, but "buttons" doesn't seem to do anything.

3.
Rather than sourcing captions from "data-caption", I would very much like to be able to pull image captions from a figcaption element on my page. Is this possible in the current build, or can we get this in an update? I'd be totally fine with having to do something like:

`caption : $(this).next("figcaption"),`

Just so long as I have an option, or some way of defining the caption location. Or maybe this is already possible using callbacks?

Finally, now that Fancybox 3 is released, would you like us to start opening new issues for this type of thing, or keep conversation going in this thread?

Thanks,
Matt

**Comment by fancyapps** [comment]
@mjcampagna Hi, thanks for your effort, I really appreciate your help!

I`ll update documentation and will include some missing parts too.

About caption: the "caption" option is somehow special, because it accepts function and return value is used as actual content of caption area.

Depending on your markup, you can use it like this - 

```
$( '[data-fancybox]' ).fancybox({
  caption : function( instance, item ) {
      return $(this).find('figcaption').html();
  }
});
```

http://codepen.io/fancyapps/pen/EZPpRG?editors=1010

**Comment by redpik** [comment]
Hi @fancyapps , I think it's urgent to create a branch for v3. some people (like me) want to help.

**Comment by mjcampagna** [comment]
@fancyapps That works perfectly for my captions. Thanks! I made just one small change:

`return $(this).closest('figure').find('figcaption').html();`

**Comment by fancyapps** [comment]
@redpik a new branch can be found there - https://github.com/fancyapps/fancyBox/tree/3.0

**Comment by robfaas** [comment]
I think an optional download button in the toolbar would be a useful feature.

**Comment by redpik** [comment]
yay! thank you @fancyapps !

**Comment by mjcampagna** [comment]
@robfaas 

> I think an optional download button in the toolbar would be a useful feature.

You should be able to add one easily.

Using the baseTpl option, use the default code in the documentation. Inside of <div class="fancybox-buttons">, add this:

`'<a class="fancybox-download" download>Download</a>' +`

Then set a callback as follows:

`onComplete	 : function( instance, item) { 
	$('.fancybox-download').attr('href', item.src);
} `

**Comment by robfaas** [comment]
Thanks @mjcampagna, I am going to try that solution.

**Comment by mjcampagna** [comment]
@robfaas 

Also, if you'd rather not mess with the baseTpl option, you can add the button via callback:

`onInit       : function( instance, item) {
	$("<a class='fancybox-download' download>Download</a>").appendTo( ".fancybox-buttons" );
}
`

That'll save you some bytes of code if you're not otherwise changing the template.

**Comment by fancyapps** [comment]
@mjcampagna be careful with that, if there are multiple instances on the page, then your code would append button to all of them. There is a handy collection of commonly used elements : 
`instance.$refs` and element containing buttons would be `instance.$refs.buttons`.

There are two possible solutions to create a download button.

1) Create `<button>` element and use click event to force downloading current image: http://codepen.io/fancyapps/pen/QdyzWm

2) Create link and use callback to change its target:
http://codepen.io/fancyapps/pen/dNGapJ

**Comment by fancyapps** [comment]
@adam-shaun that jumping is caused by your CSS transition rule: 
`transition: all 0.4s ease-out`. Since you are animating only `top` value, you could easily fix this by changing that rule to `transition: top 0.4s ease-out;`

**Comment by fancyapps** [comment]
@d3rson It is not currently possible, but I`ll implement that.

**Comment by adam-shaun** [comment]
@fancyapps that solved it. Thanks!
@mjcampagna Hey! It might be helpful for me to know how to fix the syntax errors correctly in Dreamweaver. Any help would be most welcome. Thanks. 👍

**Comment by mjcampagna** [comment]
@adam-shaun 

Open your pages in Chrome, and press Cmd-Opt-I (on Mac) to bring up the developer tools; I'm not sure about the Windows keys, but probably Ctrl-Alt-I? Click on "Console"; errors will show up in red. You'll want to clear these up to ensure your pages are properly running scripts, etc.

On p2_test.html, it looks like you need to delete line 222 altogether. It contains only "});", which appears to be a duplicate of line 221.

a2_test.html looks clean, but d2_test.html has numerous errors before fancybox runs, and those errors are killing the script entirely.

**Comment by mjcampagna** [comment]
@fancyapps I think I've found a bug. When the modal is open, focus remains on the trigger element. After opening Fancybox, press Return on your keyboard a few times to trigger the element multiple times ... you'll open the same modal again and again, and will have to close each individually.

I can prevent this by adding this to the onInit callback, but I probably shouldn't have to:

`  	instance.$lastFocus.blur();
`

**Comment by anonle** [comment]
1) How disable zoom for images?
2) How make fancyBox to be closed instead of zooming _when clicking outside of the image_?

**Comment by fancyapps** [comment]
@mjcampagna Thanks. Fixed in the latest commit.

@anonle There are currently no options for that, but I`m thinking about adding them.

btw, any ideas about naming? Something like `closeWhenClickedOutsideOfImage : true` would be self explaining, but somehow does not look right ... :)

**Comment by Luderee** [comment]
Loving the new version so far! However, it doesn't work with filtering anymore. I use Isotope for this, and in fancybox v2 I used something like the following code to switch the group:

```
$(".filter-button-group").on("click", "div", function() {
	var filterValue = $(this).attr("data-filter");
	$iso.isotope({
		filter: filterValue
	});
	
	if (filterValue == "*") {
		$(".fancybox").attr("data-fancybox", "gallery");
	} else {
		$(filterValue).attr("data-fancybox", filterValue);
		//Update fancybox or something
		console.log("Filtered");
	}
});
```

Now, once a fancybox is opened, it sticks to that selection no matter the filtering. So if you open on pageload, it will display all pictures, even after filtering. If you first filter, then open a fancybox, it will display the correct selection. Going back to all pictures will not open a fancybox.

Probably I need to reinitialize fancybox somehow after each filter is applied. How can I accomplish this?

**Comment by fancyapps** [comment]
@Luderee Easy - just create custom click handler for your links and open fancyBox manually - http://codepen.io/fancyapps/pen/EZKYPN?editors=1010

```
$('.fancybox').on('click', function() {
  $.fancybox.open( $('.fancybox:visible') );
  
  return false;
});
```

**Comment by mjcampagna** [comment]
> btw, any ideas about naming? Something like closeWhenClickedOutsideOfImage : true would be self explaining, but somehow does not look right ... :)

bgClick: 'close'

clickBackground: 'close'

closeBackground: true

backgroundOnClick: 'close'

backgroundOnClickX: true

**Comment by nikas64527** [comment]
> btw, any ideas about naming? Something like closeWhenClickedOutsideOfImage : true would be self explaining, but somehow does not look right ... :)

May by closeClickOutside? It's not necesserily for image. And set "true" by default.

**Comment by robfaas** [comment]
Or closeOnClickOutside / closeOnClickOutsideImage

**Comment by fancyapps** [comment]
OK, `closeClickOutside` has been implemented.

**Comment by HelgVolkov** [comment]
Thanks a lot! Really good job. I'm still use jquery 2.2.4 for a lot projects (some plugins require it). And there are some problems with image zoom in opened fancybox item. If I try zoom image by click, it jumps, not zoom.

**Comment by mjcampagna** [comment]
Just sharing. I've been customizing the UI, and this positioning technique for the buttons spares having to fiddle with specific pixel values when changing the size of buttons. Whatever the button size, the icons will remain centered.

```
.fancybox-button::before,
.fancybox-button::after {
	top: 50%;
	left:50%;
	-webkit-transform: translate(-50%,-50%);
	    -ms-transform: translate(-50%,-50%);
	        transform: translate(-50%,-50%);
}

.fancybox-button--close::before {
  -webkit-transform: translate(-50%,-50%) rotate(45deg);
      -ms-transform: translate(-50%,-50%) rotate(45deg);
          transform: translate(-50%,-50%) rotate(45deg);
}

.fancybox-button--close::after {
  -webkit-transform: translate(-50%,-50%) rotate(-45deg);
      -ms-transform: translate(-50%,-50%) rotate(-45deg);
          transform: translate(-50%,-50%) rotate(-45deg); 
}

.fancybox-button--left::after {
  -webkit-transform: translate(-50%,-50%) rotate(-135deg);
      -ms-transform: translate(-50%,-50%) rotate(-135deg);
          transform: translate(-50%,-50%) rotate(-135deg);
}

.fancybox-button--right::after {
  -webkit-transform: translate(-50%,-50%) rotate(45deg);
      -ms-transform: translate(-50%,-50%) rotate(45deg);
          transform: translate(-50%,-50%) rotate(45deg);
}
```

**Comment by phifa** [comment]
@fancyapps thank you again for your great plugin. Will you make it available on bower soon? thanks!

**Comment by Reda-Ahmad** [comment]
can I put only numbers in the caption field, because it wont show anything if I do not put any alphabets.

**Comment by fancyapps** [comment]
@Reda-Ahmad Looks like you have found a bug. It will be fixed in the next commit.

**Comment by saltnpixels** [comment]
OMG!! I just discovered this!!! when's it gonna be done!!

**Comment by Luderee** [comment]
@fancyapps Thanks for the solution! Only now, since no start slide is specified, it always opens on the first image of the array. How to make it so that it opens the image which is clicked?

**Comment by fancyapps** [comment]
@Luderee I have updated the snippet:

```
// Custom click event - open fancyBox manually
$('.fancybox').on('click', function() {
  var visibleLinks = $('.fancybox:visible');
  
  $.fancybox.open( visibleLinks, {}, visibleLinks.index( this ) );
  
  return false;
});
```

**Comment by saltnpixels** [comment]
@fancyapps I love the new plugin. it works great. But is there an easy way to add more images to a gallery AFTER its been opened, so im not loading tons of images unless necessary?

**Comment by fancyapps** [comment]
@saltnpixels : Well, sometimes people ask me about this but I have never understand why would you need this and how would you use that.

Anyway, each instance has `group` object containing gallery items. You can add/remove items if you wish. You would need to call `instance.updateControls();` to refresh interface afterwards. But it depends on the usage. 

Example on appending one extra image after opening a gallery:

```
$("[data-fancybox]").fancybox({
	onInit: function( instance ) {

		instance.group.push({
			index : instance.group.length, 
			opts  : instance.opts,
			src   : 'https://placeholdit.imgix.net/~text?txtsize=33&txt=350%C3%97150&w=350&h=150',
			type  : 'image'
		});

	}
});
```
http://codepen.io/fancyapps/pen/ggwmbb?editors=1010 
In this case you would not need to update interface, because it will be created later anyway.



Maybe you are looking for something like infinite loader that would create and load prev/next gallery item after user clicks on pev/next button?  ¯\_(ツ)_/¯

**Comment by saltnpixels** [comment]
@fancyapps 

Maybe I should explain my use case. Perhaps I don't need this.

My example is this:
I have a dating site. There is a grid of users with images. When you click one user, it opens in fancybox and you can see other images of _**same**_ person (each user has his own gallery of up to 4 images). 
I don't want to load all 4 images of a person unless you click the main one in the grid.

**Comment by fancyapps** [comment]
I think you have two options
1) Open fancybox manually after user clicks on the link
2) Simply hide the rest of the group images - http://codepen.io/fancyapps/pen/KagWmX - but I think in this case the browser will load thumbs anyway.

**Comment by saltnpixels** [comment]
Im using option number two. Seems easier and Im not sure how to do option 1. Does that mean using an event to open fancybox rather than have data-fancybox on the element?

anyways with option 2, from what I see in safari under the resources tab, the images are not being loaded until you click the non hidden image.

**Comment by saltnpixels** [comment]
@fancyapps 
it wont load the other images if you dont put an img tag in the anchors.
http://codepen.io/anon/pen/MJjprG

**Comment by fancyapps** [comment]
Manual opening is very easy (for example, see example I posted above http://codepen.io/fancyapps/pen/EZKYPN?editors=1010) - just call `$.fancybox.open( your-elements, your-options );` anywhere in the code.

About image loading - it depends on the browser.

**Comment by saltnpixels** [comment]
Interesting demo!

Alright, thanks a lot. So I'll see which way I do it.
For now it looks like most browsers don't load an anchor's link even if its an image. 
And your plugin seems to make the img element, only after its been opened so thats good.
Safari and Chrome don't load it till its clicked.
I'll probably stick with hidden links for now. I'll keep this here for documentation!

Lastly on safari I had weird ajax scrolling issue with fancybox 3. I fixed by scrolling to top when the ajax loads in. Else it opened the ajax page but scrolled to bottom.
https://github.com/fancyapps/fancyBox/issues/1187

**Comment by saltnpixels** [comment]
eh...
how can I open an ajax page manually? and with a selector?

**Comment by brunushky** [comment]
Hi, I has been using this plugin almost since version 1, and I found it very helpful, thanks for your effort!!!.

Now I was playing around with version 3 and found that changing background color through "helper" option don't work, or others like thumbs size....  helpers is still working or has been deprecated??, I don't found any reference in source code too. thanks.

`    helpers:  {
        thumbs : {
            width: 10,
            height: 10
        },         overlay : {
            css : {
                'background' : 'rgba(255,255,255,0.5)'
            }
        }
    },`

**Comment by mjcampagna** [comment]
Is there some way to check the group name? I'm digging into the instance, but only seeing the array and id. If I've got something like `data-fancybox="gallery"`, I'd like some way of getting the value of "gallery".

**Comment by jarred-cz** [comment]
One bug maybe - if AJAX response contains HTML comment (etc. ""), the "close button" won't show and following error in JS console appears:
_Uncaught TypeError: Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'. (jquery-3.1.1.min.js:3)_

**Comment by saltnpixels** [comment]
@fancyapps Thanks for making this. I am having a lot of fun.
Here I have profile images and tapping just selects them (this will be used for a different functionality)

But taphold opens fancybox ajax and shows a users full profile. Swipe right and it gets thrown away!
:)

![giphy](https://cloud.githubusercontent.com/assets/11615474/21988960/22fde720-dbd7-11e6-9331-f5243263a704.gif)

**Comment by saltnpixels** [comment]
@fancyapps 
if I change the href attribute with js I lose the fancybox ability because of no delegation. Why cant fancybox have the event on document by default?
$(document).on('click', '[data-fancybox'], ....

Otherwise I have to do this myself manually.

**Comment by fancyapps** [comment]
@saltnpixels very nice effect :) 

Sorry, I did not quite understand your last comment. Take a look at the bottom of core.js - 
`$(document).on('click.fb-start', '[data-fancybox]', _run);`
Does this not work for you?

**Comment by saltnpixels** [comment]
@fancyapps 
during my code I change href attribute on a data-fancybox element. But it continues to point to open the old href value.

But you are binding it to the document? hmmm... so why wont it change.

**Comment by fancyapps** [comment]
@jarred-cz the script appends close button to the first element. I did not want to over-complicate and did not implement any checking (like, what if there is no containing element and what if the first element, for example, is heading).

**Comment by fancyapps** [comment]
@mjcampagna yeah, that value is not stored anywhere and that is correct way to get it by yourself.

**Comment by fancyapps** [comment]
@saltnpixels are you sure you are actually changing it? Are you using attr() method?

**Comment by fancyapps** [comment]
@brunushky there is currently no such property, but you can change it using CSS or use onInit callback if JS is needed.

**Comment by saltnpixels** [comment]
@fancyapps 
It works on images. It's not working when I change attr('data-src'...
For ajax...

I see the change in inspector.

**Comment by fancyapps** [comment]
@saltnpixels I have no idea, this 
`$('[data-fancybox]').attr('data-src', 'https://c2.staticflickr.com/6/5761/30508382101_fdff890a83_h.jpg');` 
works fine for me.

**Comment by saltnpixels** [comment]
@fancyapps it works until you open the fancybox. If you then close it and try to change the attribute it stops working. At least for me...

**Comment by fancyapps** [comment]
Yes, the script gathers all links at the beginning. This is because we need all of them to do some calculations and, for example, create thumbnails.

**Comment by saltnpixels** [comment]
@fancyapps 
http://codepen.io/saltnpixels/pen/pREGMM

See the data-src changes but once the link is opened, it cannot be changed. How can I fix this? Or do I have to make a manual fancybox that changes each time?

**Comment by fancyapps** [comment]
Oh, you should have started this conversation with that example :)
You can change these values like this:

```
$('button').on('click', function() { 
    $('.changeable').data('type', 'image').data('src', 'https://farm6.staticflickr.com/5591/15008867125_68a8ed88cc_m.jpg')
    
});
```

**Comment by saltnpixels** [comment]
@fancyapps :/ why does .data() work but not atrr()?

im always confused between these because sometimes attr() works and soemtimes data works.
Let me try data in my code and cross my fingers....
Works...

im officially confused

**Comment by saltnpixels** [comment]
Ok, so that works... somehow...
But here:
http://codepen.io/saltnpixels/pen/jyMJvW
You cannot change the fancybox group! :D

**Comment by brunushky** [comment]
Thanks, I found the problem... I followed the GitHub link from http://fancyapps.com/fancybox/3/, and it's linking to the **main** fancybox branch, so the instructions were for v2.x .
I think that link should point maybe to https://github.com/fancyapps/fancyBox/tree/3.0, just in case somebody like me miss this detail. :)

**Comment by peeter2** [comment]
Hello,

Very nice. Just one suggestion/bug. On mobile I can't close a youtube/vimeo video by swiping ON IT. It's only functional when I swipe on the edges (overlay) of the lightbox, not on the video itself. So it is really painful to close it.
Also on desktop I can close the lightbox by clicking on the background overlay, but on mobile it does not close. When I tap on the overlay the only thing that happens is that the X button disappears from the corner. When I tap again, it reappears.

I hope these things will be fixed, then it would be superb.

Even better if there was be an option that a back button would close the lightbox on mobile because otherwise my site gets too high bounce rate. People confuse the lightbox with a new page and try to get back but they get back to the google results page when clicking back on mobile.

**Comment by jarred-cz** [comment]
Any chance to display main buttons (slideshow, full screen, close button,...) when AJAX is loaded? It seems that "buttons: true" does nothing for AJAX/iframe slides.
But "infobar: true/false" option works even for AJAX/iframe.

**Comment by brunushky** [comment]
Hi, there is a way to create a custom button let's say fullscreen, without display the default one?, or create the Fullscreen object and in some way setup a template variable to replace the original button template on FullScreen prototype? 

Thanks.

**Comment by mjcampagna** [comment]
@brunushky The fullscreen button is in the CSS; you can easily restyle it in any way that you like.

**Comment by mdss-single** [comment]
1) padding option doesn't review in docs (but it works for me)
2) prev version have a closeClick option. What about this version?
3) I need to show preview grid automatically. How can I do it?
I tried `<a href="#" data-fancybox="gallery" data-fancybox-grid>` but it doesn't work.

**Comment by cstauf2** [comment]
@fancyapps is it possible to use swipe on mobile for a group of inline elements? if so, I can't seem to figure out how. right now I have to manually close each and open the next.

**Comment by fancyapps** [comment]
@cstauf2 Yes, see this demo - http://codepen.io/fancyapps/pen/ZLLKYo

**Comment by fancyapps** [comment]
@mdss-single 
1) Well, v3 does not apply white border around images, because nowadays everyone follows the flat design :). But for html content - you can style it using CSS. It is even better, because now you can, for example, to choose different styling for small/large screen sizes using media queries. 
2) Not yet implemented.
3) See this demo - http://codepen.io/fancyapps/pen/RKKVVa?editors=1010

**Comment by fancyapps** [comment]
@brunushky: why would you need that? Anyway, you can
1) change button design using CSS
2) use fullscreen module as a template to create your own module. 

btw, there is a FullScreen instance available, but due to browser secyrity, it is not possible to start fullscreen mode grammatically:

`$.fancybox.getInstance().FullScreen.launch();`

**Comment by fancyapps** [comment]
@jarred-cz You have to disable "smallBtn" feature to enable toolbar buttons. See this demo - http://codepen.io/fancyapps/pen/KaamGx

This works this way because otherwise there would be two close buttons simultaneously :)

**Comment by jarred-cz** [comment]
@fancyapps ahh it works! thanks man ;-)!

**Comment by peeter2** [comment]
any solution to my problem? I can't swipe away a youtube video unless I carefully touch the edges of the fancybox (overlay). Swping on the video has no effect.

**Comment by mdss-single** [comment]
@fancyapps 
Thanks, It's working for me.

**Comment by fancyapps** [comment]
@peeter2  it is not possible to implement swiping on youtube videos, because they are created inside `<iframe>` tag and you don't have control of events within the iframe due to browser security policy.

**Comment by Reda-Ahmad** [comment]
i want to customize the options but the data-fancybox="" field in my code is dynamic as i am getting the images from a database, I am trying to implement the preview grid automatically mentioned above. 

How can i use the following code but make it apply to everything;
```
$('[data-fancybox="images"]').fancybox({
  thumbs : {
    showOnStart : true
  }
})
```

**Comment by fancyapps** [comment]
@Reda-Ahmad You can set this as default value:
`$.fancybox.defaults.thumbs = { showOnStart : true };`

**Comment by Reda-Ahmad** [comment]
it is still not working and also the morphing overlay is not working, when i debugged, there were no errors but still its not morphing

**Comment by peeter2** [comment]
I see. And how about the other anomaly? Fancybox cannot be closed on mobile when clicking on the overlay. The only thing that happens is that the X disappears. Can this be fixed? On computer I can close it by clicking on the overlay but on mobile it does not work.

**Comment by fancyapps** [comment]
@Reda-Ahmad it is working for me - http://codepen.io/fancyapps/full/bggzvo/

**Comment by fancyapps** [comment]
@peeter2 that is intentionally - taping on touch-enabled devices always toggles controls.

**Comment by Reda-Ahmad** [comment]
I dont know why but its not working for me, where should I keep the piece of code.

**Comment by fancyapps** [comment]
@Reda-Ahmad Have you updated to the latest version? Put that code anywhere after jQuery and fancybox js files.

**Comment by peeter2** [comment]
Nice... then I have only one request. I want the mobile back button to close the fancybox. I can pay some beers if anyone can edit the code to achieve this.
Currently it is bad for SEO because visitors are taken back to google search results when they click back.

**Comment by Reda-Ahmad** [comment]
I updated the files, but it is still not working, I tried changing the animation speed using the same format and it worked, but thumb preview is not showing automatically.

**Comment by fancyapps** [comment]
@Reda-Ahmad Well, then show your page, otherwise I have no clue. If you are using the latest code and also that snippet, then it should work. 

@peeter2 There you go - https://github.com/fancyapps/fancyBox/commits/3.0

**Comment by Reda-Ahmad** [comment]
How can I show my page? I am using dreamweaver.

**Comment by peeter2** [comment]
Thanks. However now it just seems like the backspace key works on desktop (but not on mobile). On mobile the back button still takes me back to the previous page.

**Comment by fancyapps** [comment]
You mean - you want to change behavior of physical device button/browser navigation button? You can warn user using "beforeunload" callback and that`s it.

**Comment by peeter2** [comment]
Oh right. I see. I could also try some kind of workaround such as changing the address bar with ajax so the fancybox makes the browser think it's a new page. But that's more work I guess so I should hire a developer.

**Comment by fancyapps** [comment]
@peeter2 What you are looking for is not possible - that would be a huge security problem. See https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload

**Comment by Reda-Ahmad** [comment]
I updated the fancybox files and cleared the browser cache, the thumb preview is now showing automatically but I get an empty placeholder instead of the images when opened, none of the images are showing but their thumbnails still show. I reverted to the old fancybox box files to check and the images worked. Something in the new version is messing up the images.

**Comment by fancyapps** [comment]
I have tested just now on:

- Windows 10:
    Chrome - works
    FireFox - works
    Edge - works
    IE 11 - works
    IE 10 - works
    IE 9 - works

- MacOS Sierra
    Chome - works
    FireFox - works

- iPad
    Safari - works
   Chrome - works
    
But you are arguing that it does not work...

**Comment by Reda-Ahmad** [comment]
check this screenshot, the new version is not showing the images, but the old one did. I just updated the files and it stopped working. [https://snag.gy/THt0lR.jpg](url)

**Comment by mjcampagna** [comment]
@Reda-Ahmad You haven't provided a working link, so we can't see what you're actually doing, but this works fine for me:

```javascript
			onActivate : function( instance, current ) {				
				instance.Thumbs.show();
			},
```

If that's not working for you, then you probably have an error elsewhere in your implementation, and seeing your code running would go a long way toward seeking a solution.

**Comment by vick80** [comment]
I use FancyBox 2 together with Cloud Zoom (the enlarged image in FancyBox by clicking opens):
http://codepen.io/ethan2/pen/zNZjmP
After the FancyBox updating to the 3rd version, the image ceased to boot, only their total quantity is correctly displayed:
http://codepen.io/ethan2/pen/QdprBy
Prompt what changes need to be made in my code that everything earned?

**Comment by fancyapps** [comment]
@vick80 the problem is that `cloudZoom.getGalleryList()` returns content type as "iframe" for gallery items. You could either build your own list or ask author to update his plugin.

**Comment by vick80** [comment]
@fancyapps Thanks!

**Comment by Reda-Ahmad** [comment]
I am having a small issue, when reducing the window side the buttons behave normally, but if I reduce the window size with the thumb preview open and the window reaches mobile size, the close button doesnt stick to the top right.

**Comment by fancyapps** [comment]
@Reda-Ahmad I think this issue was fixed in one of the previous commits.

**Comment by Reda-Ahmad** [comment]
@fancyapps I figured out why the image was not opening on the latest build, it is because my image url links have some spaces inside. I tested it out with a non space url and it worked. It is going to be a lot of work to change the urls of over 300 images but maybe that is something which you can fix.

**Comment by fancyapps** [comment]
@Reda-Ahmad Looks like you have to specify content type, add `data-type="image"` to your links.

**Comment by Reda-Ahmad** [comment]
@fancyapps still not working with the space urls.

**Comment by fancyapps** [comment]
@Reda-Ahmad http://stackoverflow.com/questions/4172579/html-href-syntax-is-it-okay-to-have-space-in-file-name

**Comment by Reda-Ahmad** [comment]
@fancyapps Thanks, I was already writing the php code to convert the spaces to %20.

**Comment by mplunkett5** [comment]
Really excited for this release! It looks immense!

**Comment by JU5TY** [comment]
Looks great, desperate to use this via Bower on an upcoming project, any ideas when we can expect that?

**Comment by catonkatonk** [comment]
I have an issue when showOnStart  is set to true:
If I scroll down the page, then click the picture to launch Fancybox, the picture comes in low, below where it should be. Everything else is positioned normally. Clicking the picture before scrolling shows it where it should be. Scrolling back up and then clicking also positions the picture fine.
I reset everything back to default - default CSS, no other options apart from showonstart=true selected, and the problem persists. Removing that setting fixes it, obviously.

Another issue with showOnStart @ true is that it has no effect if the thumbnail button is hidden. I want the thumbnails to always display, without allowing the user to toggle them off, ideally.

**Comment by fancyapps** [comment]
@catonkatonk 
hmm, I cannot reproduce an issue with positioning.  Could you show some screenshot? It would be ideal if you could create a screencast.

I`m thinking about creating an option like "createButton: false" to make it work without a button.

**Comment by catonkatonk** [comment]
Here you go:
http://screencast-o-matic.com/watch/cbVT2aQsSp
This was in Opera, but Edge and IE also exhibit the same issue.

If it helps, I'm on jQuery 1.12.4 (sigh, Wordpress), wonder if that has anything to do with it.

**Comment by catonkatonk** [comment]
jQuery 1.12.4 is indeed the issue.

I took this Codepen:
http://codepen.io/fancyapps/pen/oBedNj
Stuck in this CSS: p {margin-bottom: 200vh}
And swapped out jQuery 3 for 1.12.4
And scrolled down, clicked a thumbnail while it was near the top of the frame, and could reproduce the issue.

Do I have to just suck it up and upgrade jQuery?

**Comment by nrkhatana** [comment]
unable to set iframe width, can anyone explain bit of sample code plz?

**Comment by fancyapps** [comment]
@catonkatonk The last commit should fix that.

@nrkhatana You have two options:

1. Using JS:

```
$('[data-fancybox]').fancybox({
  iframe : {
    css : {
      width: '500px'
    }
  }
});
``` 

2. Using CSS:

```
.fancybox-content {
  width: 500px;
}

```

**Comment by nrkhatana** [comment]
@fancyapps  Wonderful, it worked, another small issue is, I am unable to show the close button in fancybox, I have tried the following

`	$('[data-fancybox]').fancybox({
		
	  iframe : {
		css : {
		  width: '500px'
		},
		closeBtn   : true,
  		closeTpl : '<button data-fancybox-close class="fancybox-close-small">×</button>',

	  }
	});`

**Comment by fancyapps** [comment]
@nrkhatana Maybe this demo helps - http://codepen.io/fancyapps/pen/zNdWay
Custom video dimensions, navigation arrows and close button.

**Comment by GeorgeRT** [comment]
Oops, being new to this, I created a new issue instead of commenting here.  I noticed the fancybox-lock & fancybox-margin class/toggles are not implemented.  I realized I can define the class myself & do the toggle in the afterLoad/beforeClose events but would rather it be built-in?  Thanks.

**Comment by rafu1987** [comment]
Great work @fancyapps - I love the new look and features. Any plans when it's going to be stable? :-)

**Comment by bigfraggle** [comment]
I'm loving v2 and it seems v3 will be just as well! :)
Good work @fancyapps 

Will there be a mousewheel support? Desktop usability could be better...

**Comment by brunushky** [comment]
@fancyapps Hi, ok I do great updates on button theming, also including Material icons, thanks.

But I have only two more question, there is an easy workaround to include caption at bottom of the image div ('fancybox-image') ... And the same for thumbs, I already read something about thumbnails, do you think it could be to complex to arrange them in line at bottom of the page. Or should I deal with JS and build my own methods???

Thanks.

**Comment by mjcampagna** [comment]
@brunushky 

It wasn't easy, but I've been away from coding for a few days, and needed an exercise to get my head back into things, so here's a workaround to put captions at the bottom of the image. May not be perfect, but you can try it out.

The CSS:

```css
.fancybox-caption-wrap,
.fancybox-show-caption .fancybox-caption-wrap {
	opacity:0;
	transition: 0.2s opacity ease-in-out 0.4s;
	visibility: hidden;
}

.floating-caption {
	background-color: transparent;
	color: #fff;
	font-size: 0.875rem;
	margin: 0;
	padding:0;
	opacity:0;
	position: absolute; top: 50%; left: 50%;
	transition: 0.2s opacity ease-in-out 0.4s;
	visibility: hidden;
	width: 100%;
	z-index: 99995;
}

.fancybox-show-caption.fancybox-default-caption .fancybox-caption-wrap,
.fancybox-show-caption.fancybox-float-caption .floating-caption {
	opacity: 1;
	visibility: visible;
}

.floating-caption p {
	margin: 0;
}
```

The Javascript:

```javascript
$('[data-fancybox=gallery]').fancybox({

	beforeMove : function( instance, current ) {

		instance.$refs.container.removeClass('fancybox-default-caption fancybox-float-caption');

		var caption     = $('<p></p>').text(current.opts.caption);

		instance.$refs.floatingCaption = $('<div class="floating-caption"></div>').html(caption)
			.appendTo(current.$slide);

	},

	onComplete : function( instance, current ) {

		instance.$refs.floatingCaption.css({
				maxWidth : current.width,
				width : current.width,
				transform : 'scale(1.0) translateX( -50% ) translateY( '+ ( current.height/2 + 6 ) +'px )',
			})

		var captionHeight = instance.$refs.floatingCaption.height(),
		    captionOffset = instance.$refs.floatingCaption.offset().top - $(document).scrollTop(),
		    marginBottom  = window.innerHeight - captionOffset
		  ;

		if ( marginBottom >= captionHeight + 12 ) {
			instance.$refs.container.addClass('fancybox-float-caption');
		}
		else {
			instance.$refs.container.addClass('fancybox-default-caption');
		}

	},

});
```

This creates a second container for the caption, the floating caption. If there's room enough beneath the image, it will display the floating caption beneath the image. If there is not room enough, then it will show Fancybox's default caption instead.

**Comment by jarred-cz** [comment]
@fancyapps any plans to support localization? some text can be changed if I manualy define template and pass the translated string into it (eg. "Close (X", "Next", "Previous"), but some string (eg. "Thumbnails (G)", "Slideshow (P)" etc.) are hardcoded in JS source code.

**Comment by fancyapps** [comment]
@brunushky The option to display caption below image is not yet implemented. I have not found a nice solution that would play well with the overall structure. Because in such case one block dimensions would depend on the other one.

But, looking from the bright side, the layout has simple structure, therefore you can easily move or resize each block, just like playing with Lego.

I have created two demos for displaying thumbnails at the bottom of the page:

[Version #1](http://codepen.io/fancyapps/pen/vgRajd/) - Full size image has limited area.

[Version #2](http://codepen.io/fancyapps/pen/zNWLoG) - Full size images goes below thumbnails.

@jarred-cz @bigfraggle These features are in the todo list.

**Comment by stephangrass** [comment]
Hi,
great work! 
When using html, body { height: 100%; }, the bodycontent will jump to top, when opening a lightbox. After closing, the content will jump back to the right position.

p.s. Will you offer fancybox 3 by the same licensing options as fancybox 2?

**Comment by fancyapps** [comment]
@stephangrass Thanks.

I see no jumping - http://codepen.io/fancyapps/pen/QdmPod?editors=1100

Yes, licensing will be +/- the same. I'm thinking about setting the price a bit higher but include support.

**Comment by stephangrass** [comment]
Hmm. So I have to search a little bit. In my markup html, body { height: 100%; } will force jumping

**Comment by fancyapps** [comment]
@stephangrass Could you show your page? Maybe you have horizontal scrollbar?

**Comment by stephangrass** [comment]
It's a local installation.

I have set overflow-x: auto and overflow-y: auto to the html

Setting overflow-y: auto; and height: 100% to the html-element will  force jumping

**Comment by fancyapps** [comment]
@stephangrass Why would you need these CSS rules? 

Anyway, the last commit should fix that. The demo seems to be working fine, at least - http://codepen.io/fancyapps/pen/QdmPod?editors=1100

**Comment by stephangrass** [comment]
I used this in a special mediaquery-context.
Thanks.

**Comment by juanram0n** [comment]
@fancyapps just only a little detail. On the web, on the Setup section, the css it´s not the minimized file and could be confused:

`<link rel="stylesheet" type="text/css" href="jquery.fancybox.css">`

I´m very excited with the new version, thanks a lot!

**Comment by Tuning-Soft** [comment]
Excellent work, It looks really cool!!!

**Comment by brunushky** [comment]
@fancyapps, @mjcampagna Thanks, that are great advises to start!!!... I will let you know the final results.

**Comment by Tuning-Soft** [comment]
@fancyapps  
I need more zhogova :))

May be You can add button and html-block for comments? 
Some, Like this:
https://monosnap.com/file/X7uTNUMJGWP48vrAMIxYJNdy68eA74.png

Not comments, only html-block for include comments from facebook, discuss, ajax, etc.
Now i'm very need this, i can pay commerce lincense for my eCommerce module, it's will be very cool!!!

End how change picture id in adress bar, for ajax get comments from DB to this picture?
https://monosnap.com/file/pV5khi9ATaDrASRenWcHTKinVT0oU5.png

**Comment by Tuning-Soft** [comment]
And "Share button" very good idea!
![2017-02-05_04 42 59](https://cloud.githubusercontent.com/assets/1108013/22621766/b1e65554-eb5d-11e6-8623-e2a8242ab4f8.png)

**Comment by fancyapps** [comment]
Hi, Everyone!

I am very sorry, I got sick and therefore there will be some delay.

Thanks!

**Comment by brunushky** [comment]
@fancyapps  Take care, and fast recovery!!!

**Comment by fancyapps** [comment]
OK, as you might have noticed, there has been some work lately and I think that v3 is ready to be launched!
I`ll update docs about recent changes (now supporting "srcset", for example), make final changes to homepage and then finally switch branches to 3.0.

**Comment by redpik** [comment]
:+1:

**Comment by leecollings** [comment]
https://github.com/fancyapps/fancyBox/issues/1239

**Comment by erickmonteiro** [comment]
👍

**Comment by VieiraSA** [comment]
Why is fancybox 3 duplicating images ( Please see the attached screenshot )? So it displays wrong numbers on the interface in the fancybox-infobar element.
How could I fix that?

![screen shot 2017-05-17 at 7 03 17 pm](https://cloud.githubusercontent.com/assets/6270644/26178245/922e26da-3b33-11e7-9eff-5b619ee5f96a.png)

**Comment by fancyapps** [comment]
@VieiraSA Sorry, no idea what are you talking about. https://github.com/fancyapps/fancybox#bugs-and-feature-requests

**Comment by bharatparmar** [comment]
i am using fancybox 3 and in my page there is gallery of 6000+ images. when i click on top thumb button my browser becomes unresponsive. is there is a way to make thumb load more like option.

**Comment by YuliyaYakutsik** [comment]
On mobile devices when I click on overlay the fancybox doesn`t close. On desktop it runs, when I click on overlay the fancybox closes. Can I fix this problem?

**Comment by fancyapps** [comment]
You can "fix this problem" by using `mobile` option, this is default value:

```JS
mobile : {
    clickSlide : function( current, event ) {
        return current.type === 'image' ? 'toggleControls' : "close";
    }
}
```

You can change it to your liking.

**Comment by YuliyaYakutsik** [comment]
Thank you very much! Now it works, I`m happy!

**Comment by philippezumofen** [comment]
really great lightbox! Looks amazing and acts perfectly.

But: I'm looking for a possibilty to show the images in the lightbox not chronologically (as they appear in the html-doc), but individually (with an attribute like "data-order" or somthing similar (or a css class)). Unfortunately my javascript skills are very low and i can't do this on my own...

I found a codefragment working for fancybox 2 (https://stackoverflow.com/questions/38650690/reorder-fancybox-items-in-gallery ) but none for fancybox 3.

The site, where I want to assign the order of the images is: www.philippezumofen.ch (the galleries consists of 2 parallel divs and is not a javascript-generated grid, thats why i need the order-option). As the website consists of 46 sites a change to a javascript-generated grid is not an option, because i would have to change the whole structure of every site.

Is this individual ordering somehow possible?

Thank you for a response

**Comment by fancyapps** [comment]
@philippezumofen First, you have to understand how fancyBox works. 

fancyBox depends on jQuery, but jQuery plugin is just a tiny part of the whole code. When you execute `$('.selector').fancybox()`, you are calling `$.fn.fancybox` method. And what this method does - it just  attaches click handler to selected elements and nothing more. Then this click handler receives clicked element, finds group, index and then calls `$.fancybox.open( items, opts, index )` where actual magic happens. 

What that means is, that you can create your own click handler. That is exactly what is being done in the sample that you provided.

**Comment by jock78** [comment]
Hi i got one issue with fb3 - when im closing gallery my entire container get display: NONE style and nothing is visible on site.. how to prevent it to happen?

**Comment by fancyapps** [comment]
@jock78 Your question could be quoted as "How not to ask a question".

`A good bug report shouldn't leave others needing to chase you up for more information. Please try to be as detailed as possible in your report.`

https://github.com/fancyapps/fancybox#bugs-and-feature-requests
